frame/collective/src/lib.rs

```diff
@@ -684,19 +690,46 @@ decl_module! {
 			}
 			let approved = yes_votes >= voting.threshold;
 
-			let (p, len) = Self::validate_and_get_proposal(
-				&proposal,
+			let (proposal, len) = Self::validate_and_get_proposal(
+				&proposal_hash,
 				length_bound,
 				proposal_weight_bound
 			)?;
-			Self::deposit_event(RawEvent::Closed(proposal, yes_votes, no_votes));
-			let finalize_weight = Self::finalize_proposal(approved, seats, voting, proposal, p);
+			Self::deposit_event(RawEvent::Closed(proposal_hash, yes_votes, no_votes));
+			let finalize_weight = if approved {
+				Self::do_approve_proposal(seats, voting, proposal_hash, proposal)
+			} else {
+				Self::do_disapprove_proposal(proposal_hash)
+			};
```

Comment:
The semantics here are different from the ones above. AFAICT this code has the old behavior (of getting stuck for overweight proposals) if the proposal exits the voting period.
You'll want to reorder like so (pseudo-code):
```
let w = if approved {
    let proposal = validate_and_get(hash)?;
    do_approve(proposal)
} else {
    do_disapprove(hash)
};
deposit_event(Closed);
```
I would suggest adding a test for the new behavior after end of the voting period as well.

Reply:
Take a second look please!